### ISSUE TITLE
Excluding ArrowFunctionExpression in CallExpressions reporting as errors

### ISSUE THREAD

**Issue by martiensk** [gajus/eslint-plugin-jsdoc#1195]
## Expected behavior
When using the rule _jsdoc/require-jsdoc_ with contexts with _:not(CallExpression) > ArrowFunctionExpression_, the plugin should not trigger an error on Vue lifecycle hooks (or setTimeout).

## Actual behavior
An error is triggered.

## ESLint Config

```js
"jsdoc/require-jsdoc": [
  "error", 
  {
    "contexts": [
      ":not(CallExpression) > ArrowFunctionExpression"
    ],
    "require": {
      "ArrowFunctionExpression":false,
      "FunctionDeclaration": true,
      "MethodDefinition": true,
      "FunctionExpression": true
    }
  }
]
```

## ESLint sample

```js
setTimeout(() => {}, 10);
// Or in the case of Vue
onMounted(() => {
    initScrollTriggerAnimation();
});
```

## Environment

- Node version: 18.18.0
- ESLint version 8.53.0
- `eslint-plugin-jsdoc` version: 48.0.4

**Comment by brettz9** [comment]
I am not able to replicate getting any errors for this.

Feel free to try to replicate against https://eslint-online-playground.netlify.app/#eNp1U8FOAyEQ/RXCqTWtq9c1mhirN+PBxot4QJhWlB3WAbTG7L8Ly65tjeWwgZnHmzfM22/uSVWwkU1r4fjV85pXRwy8NRjm2nj5bIGhm0fUsGJHlUCBHsLSNOBimEym7PyCfXczdnoyPRNYVeyOmEEWXoAp6YG5FXuIINDhrYsYQI93BLK0DJpwr8hZuySzXgNdomlkMA4nma/LHz7jx0URqSTRYRLZXxccNgFQe8Fr9phOPagmUK5pUhy04LOEam1cG6xfvXaq2ss+zQpPK8kD3bW5cM82yEucqpEPQD4lclxwKwP4kIhHhHeRFCy/WiiAxulo0z7nu4GfUmSfdxTzHg3BvD+NXRA5ysIHbEYrl55uE4ZGh3BO1OjC5Epae71pCXyWOWUX7DJxfN5EVLmhbaqIyqs0XjgGETvySvwgSb2S1sMvQ8aOsAUoK6kfYOYLFPdxtxBenF7AKg/+EOjfmj1sRHVl0z31j5x8kjzSSvUm19nEuw7R8LGANo8blfkzg+KXA2Md/oHine2Atsjf0t0P+MkQTg==

**Comment by martiensk** [comment]
@brettz9 I managed to fix my problem - it is not a bug with eslint. It is a bug with my knowledge.
Below is what caused the problem, I fixed it by changing my contexts string to `:not(CallExpression):not(Property) > ArrowFunctionExpression`

```js
"jsdoc/require-jsdoc": [
      "warn", 
      {
        "contexts": [
          ":not(CallExpression) > ArrowFunctionExpression",
          ":not(Property) > ArrowFunctionExpression"
        ],
        "require": {
          "ArrowFunctionExpression":false,
          "FunctionDeclaration": true,
          "MethodDefinition": true,
          "FunctionExpression": true
        }
      }
    ]
```
